crates/orderbook/src/solver_competition.rs

```diff
@@ -11,6 +11,7 @@ use {
 pub enum Identifier {
     Id(AuctionId),
     Transaction(H256),
+    OrderUid(OrderUid),
```

Comment:
I'm not a fan of adding a non unique identifier here (there is only one competition per ID/tx_hash, but there can be many for an order uid)